compat/winansi.c

```diff
@@ -594,6 +594,35 @@ static void detect_msys_tty(int fd)
 
 #endif
 
+static HANDLE stdout_handle;
+static DWORD stdout_console_mode;
+
+static void reset_stdout_console_mode(void)
+{
+	SetConsoleMode(stdout_handle, stdout_console_mode);
+}
+
+static int enable_virtual_processing(void)
+{
+	stdout_handle = GetStdHandle(STD_OUTPUT_HANDLE);
+	if (stdout_handle == INVALID_HANDLE_VALUE)
+		return 0;
+
+	if (!GetConsoleMode(stdout_handle, &stdout_console_mode))
+		return 0;
+
+	if (stdout_console_mode & ENABLE_VIRTUAL_TERMINAL_PROCESSING)
+		return 1;
+
+	if (!SetConsoleMode(stdout_handle,
+			    stdout_console_mode |
+			    ENABLE_VIRTUAL_TERMINAL_PROCESSING))
```

Comment:
The docs for `SetConsoleMode()` say 

>Ensure ENABLE_PROCESSED_OUTPUT is set when using this flag.

so this should probably be

```c
	if (!SetConsoleMode(stdout_handle,
			    stdout_console_mode |
			    ENABLE_PROCESSED_OUTPUT |
                ENABLE_VIRTUAL_TERMINAL_PROCESSING))
```

Reply:
Good catch!

Reply:
Fixed in cf15942e384fe0d53268ed0ba98cef2ef9e1d9fb.